test/UpdateDeprecations.lean

```diff
@@ -0,0 +1,39 @@
+import Mathlib.Tactic.UpdateDeprecations
+
+run_cmd
+  let fn := "True.hish"
+  let s1 := ".hish and more text"
+  let s2 := "hish and more text"
+  let _s3 := "ish and more text"
+  if findNamespaceMatch fn s1 != some ".hish" then Lean.logInfo "error"
+  if findNamespaceMatch fn s2 != some "hish"  then Lean.logInfo "error"
+  --if findNamespaceMatch fn _s3 != none         then Lean.logInfo "error"
```

Comment:
Do you think it's possible to do some more demonstrative testing in a branch that will not be merged but is for testing purposes only?

Reply:
To test this, you can create a new file (say `Mathlib/test.lean`) containing
```lean
theorem not_useful (t : True) : True := t

@[deprecated not_useful] theorem True.lhish (t : True) : True := t

@[deprecated True.intro] theorem True.hish : True := .intro

example : True ∧ True := by
  refine ⟨?_, ?_⟩
  exact True.lhish True.hish
  exact True.hish
```
and then run
```bash
lake build Mathlib.test
lake exe update_deprecations --mods Mathlib/test.lean
```
to see the deprecations disappear.

Reply:
@jcommelin, [here](https://github.com/leanprover-community/mathlib4/actions/runs/9393230408/job/25868928673?pr=13543) you can see a test.

* I added to files with deprecated lemmas to mathlib
* I modified CI to
* * first run a plain `lake build` so you can see the warning
* * then run the `lake exe update_deprecation` to fix the deprecations
* * then run again `lake build` but flagged to fail on warning.

You can see that the full CI was successful.